content/chainguard/chainguard-enforce/chainguard-enforce-kubernetes/maximum-image-age-policy.md

```diff
@@ -0,0 +1,78 @@
+---
+title: "Limit the “Build Horizon” of Build Artifacts with the Maximum Image Age Policy"
+type: "article"
+description: "Understanding Identity and Access Management in Chainguard Enforce"
+date: 2023-10-04T15:22:20+01:00
+lastmod: 2023-10-04T15:22:20+01:00
+draft: false
+images: []
+tags: ["policy-controller", "Policies", "Enforce", "Product"]
+menu:
+  docs:
+    parent: "chainguard-enforce-kubernetes"
+weight: 100
+toc: true
+---
+
+The age of your build artifacts can have serious maintenance and security implications for your Kubernetes workloads. Stale software incurs technical debt and is often more likely to contain vulnerabilities that can put your workloads at risk. Additionally, as we discuss in our article [The Principle of Ephemerality](https://www.chainguard.dev/unchained/the-principle-of-ephemerality), refreshing your build artifacts frequently makes it difficult for an attacker to maintain a toehold in your software should they successfully breach your build system without your knowledge.
+
+The importance of freshness applies to a range of software, such as “off-the-shelf” components like monitoring tools, dependencies you compile into your software like base images, and the software you develop and maintain yourself. That said, tracking the age of every piece of software you use in your workloads can be a daunting task. As [even Chainguard’s own security-focused engineers discovered](https://www.chainguard.dev/unchained/conquer-your-build-horizon-with-chainguard-enforce-in-2023), stale software can slip through the cracks despite best efforts.
+
+## Maximum Image Age Policy
+
+One way to address this problem is to impose a “build horizon” on software artifacts. Coined at Google, a build horizon refers to the maximum age allowed for build artifacts. Here at Chainguard, we’ve created a Maximum Image Age policy that automatically checks the age of build artifacts in your Kubernetes workloads. You can use this policy to fail images or raise warnings based on their age.
+
+Written in rego (to take advantage of rego’s time functions), this policy imposes a maximum age of 30 days on all images and can be customized for other age limits. The age is measured using the container image's configuration, which has a "created" field. Note that some build tools may fail this check because they build reproducibly, and use a fixed date (e.g. the Unix epoch) as their creation time, but many of these tools support specifying SOURCE_DATE_EPOCH, which aligns the creation time with the date of the source commit.
```

Comment:
```suggestion
Written in [Rego](/chainguard/chainguard-enforce/chainguard-enforce-kubernetes/chainguard-enforce-rego-policies/) (to take advantage of Rego’s time functions), this policy imposes a maximum age of 30 days on all images and can be customized for other age limits. The age is measured using the container image's configuration, which has a "created" field. Note that some build tools may fail this check because they build reproducibly, and use a fixed date (such as the Unix epoch) as their creation time, but many of these tools support specifying `SOURCE_DATE_EPOCH`, which aligns the creation time with the date of the source commit.
```